### PR TITLE
fix: optional enum type and nullish handling

### PR DESCRIPTION
**Pull request by kpranoto-aws**
*Changes:*
* nullish string handling for our JS customers
* optional enum type to use for formatting args

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by SwaySway — APPROVED** [review]

**Review by hein-j — APPROVED** [review]

File: packages/codegen-ui/lib/types/string-format.ts [hunk 1]
```diff
@@ -14,6 +14,21 @@
   limitations under the License.
  */
 
+export enum DATE {
```
Comment (cwoolum) [inline packages/codegen-ui/lib/types/string-format.ts:17]: Can we please not use enums? It adds more work when we have to deserialize. I'd rather we just have a type that represents the available string options.

File: packages/codegen-ui/lib/utils/string-formatter.ts [hunk 2]
```diff
@@ -19,6 +19,10 @@ import { DateFormat, DateTimeFormat, TimeFormat } from '../types';
 const invalidDateStr = 'Invalid Date';
 
 export function formatDate(date: string, format: DateFormat['dateFormat']): string {
+  if (date === undefined || date === null) {
```
Comment (cwoolum) [inline packages/codegen-ui/lib/utils/string-formatter.ts:22]: `if(!date)` would cover all cases here.

File: packages/codegen-ui/lib/utils/string-formatter.ts [hunk 3]
```diff
@@ -48,6 +52,10 @@ export function formatDate(date: string, format: DateFormat['dateFormat']): stri
 }
 
 export function formatTime(time: string, format: TimeFormat['timeFormat']): string {
+  if (time === undefined || time === null) {
```
Comment (cwoolum) [inline packages/codegen-ui/lib/utils/string-formatter.ts:55]: `if(!time)` would cover all cases here.

File: packages/codegen-ui/lib/utils/string-formatter.ts [hunk 4]
```diff
@@ -79,6 +87,10 @@ export function formatTime(time: string, format: TimeFormat['timeFormat']): stri
 }
 
 export function formatDateTime(dateTimeStr: string, format: DateTimeFormat['dateTimeFormat']): string {
+  if (dateTimeStr === undefined || dateTimeStr === null) {
```
Comment (cwoolum) [inline packages/codegen-ui/lib/utils/string-formatter.ts:90]: `if(!dateTimeStr )` would cover all cases here.

**Pull request merged**